Bottom: 5b8ec8160f6bf6c8e4811e199d91c8e369fa98e3
Top:    213384a599e54d86b40b97a62f436c11c3b37c3d
Author: Roland Stigge <stigge@antcom.de>
Date:   2012-03-29 23:31:15 +0200

Fix i2c start on PNX

This patch fixes i2c start by checking for the right mif.mode

Signed-off-by: Roland Stigge <stigge@antcom.de>


---

diff --git a/drivers/i2c/busses/i2c-pnx.c b/drivers/i2c/busses/i2c-pnx.c
index e9d66b6..e4519d9 100644
--- a/drivers/i2c/busses/i2c-pnx.c
+++ b/drivers/i2c/busses/i2c-pnx.c
@@ -137,21 +137,23 @@ static int i2c_pnx_start(unsigned char slave_addr,
 		return -EINVAL;
 	}
 
-	/* First, make sure bus is idle */
-	if (wait_timeout(alg_data)) {
-		/* Somebody else is monopolizing the bus */
-		dev_err(&alg_data->adapter.dev,
-			"%s: Bus busy. Slave addr = %02x, cntrl = %x, stat = %x\n",
-			alg_data->adapter.name, slave_addr,
-			ioread32(I2C_REG_CTL(alg_data)),
-			ioread32(I2C_REG_STS(alg_data)));
-		return -EBUSY;
-	} else if (ioread32(I2C_REG_STS(alg_data)) & mstatus_afi) {
-		/* Sorry, we lost the bus */
-		dev_err(&alg_data->adapter.dev,
-		        "%s: Arbitration failure. Slave addr = %02x\n",
-			alg_data->adapter.name, slave_addr);
-		return -EIO;
+	if (alg_data->mif.mode == 0) {
+		/* First, make sure bus is idle */
+		if (wait_timeout(alg_data)) {
+			/* Somebody else is monopolizing the bus */
+			dev_err(&alg_data->adapter.dev,
+				"%s: Bus busy. Slave addr = %02x, cntrl = %x, stat = %x\n",
+				alg_data->adapter.name, slave_addr,
+				ioread32(I2C_REG_CTL(alg_data)),
+				ioread32(I2C_REG_STS(alg_data)));
+			return -EBUSY;
+		} else if (ioread32(I2C_REG_STS(alg_data)) & mstatus_afi) {
+			/* Sorry, we lost the bus */
+			dev_err(&alg_data->adapter.dev,
+				"%s: Arbitration failure. Slave addr = %02x\n",
+				alg_data->adapter.name, slave_addr);
+			return -EIO;
+		}
 	}
 
 	/*
diff --git a/sound/soc/codecs/uda1380.c b/sound/soc/codecs/uda1380.c
index 197463a..bddaf4d 100644
--- a/sound/soc/codecs/uda1380.c
+++ b/sound/soc/codecs/uda1380.c
@@ -161,13 +161,10 @@ static int uda1380_reset(struct snd_soc_codec *codec)
 		data[1] = 0;
 		data[2] = 0;
 
-		printk("JDS reset 1\n");
 		if ((temp = codec->hw_write(codec->control_data, data, 3)) != 3) {
-			printk("JDS reset 2 %d\n", temp);
 			dev_err(codec->dev, "%s: failed\n", __func__);
 			return -EIO;
 		}
-		printk("JDS reset 3\n");
 	}
 
 	return 0;
diff --git a/sound/soc/soc-core.c b/sound/soc/soc-core.c
index 306a577..92cee24 100644
--- a/sound/soc/soc-core.c
+++ b/sound/soc/soc-core.c
@@ -22,8 +22,6 @@
  *   o Support TDM on PCM and I2S
  */
 
-#define DEBUG
-
 #include <linux/module.h>
 #include <linux/moduleparam.h>
 #include <linux/init.h>
